### PR TITLE
[fsdp] fix: move CPU tensors to GPU in per_tensor_param before rollout sync

### PR DESCRIPTION
**Pull request by cavities12**
### What does this PR do?

Fix CPU tensors being passed to rollout engine without moving to GPU. `collect_lora_params()` returns CPU tensors via `.detach().cpu()`, but the `base_sync_done` fast-path in `rollout_mode()` passed them through as-is. This causes errors in backends that expect GPU tensors (e.g. SGLang's CUDA IPC serializer raises `IndexError` on CPU tensor pickle tuples).

### Checklist Before Starting

- [x] Search for similar PRs. Paste at least one query link here: https://github.com/verl-project/verl/pulls?q=is%3Apr+per_tensor_param+cpu
- [x] Format the PR title as `[{modules}] {type}: {description}`

### Test

- Existing GPU unit tests pass (no behavioral change for GPU tensors)
- LoRA + FSDP2 + SGLang rollout sync works without IndexError
- LoRA + FSDP2 + vLLM rollout sync still works (regression check)

### API and Usage Example

No API changes. Internal fix only.

### Design & Code Changes

**`verl/workers/fsdp_workers.py`** — `rollout_mode()`:

Unify `per_tensor_param` construction to handle all three tensor types consistently, regardless of `base_sync_done`:
- **DTensor**: `.to(device, non_blocking=True).full_tensor()` (async safe, already on GPU)
- **CPU tensor**: `.to(device, non_blocking=False)` (synchronous — async on non-pinned memory silently falls back to sync anyway, and previously caused garbage output)
- **GPU tensor**: pass-through

Previously, when `base_sync_done=True`, the fast-path skipped `.to(device)` entirely, passing CPU tensors from `collect_lora_params` directly to the rollout engine.

### Checklist Before Submitting

- [x] Read the [Contribute Guide](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md).
- [x] Apply [pre-commit checks](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [x] Add / Update [the documentation](https://github.com/volcengine/verl/tree/main/docs).
- [x] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/volcengine/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: This is a 1-line behavioral fix in the weight sync hot path. The bug only manifests with real FSDP + LoRA + SGLang on multi-GPU, which requires the full training loop. Covered by the integration test in #5535.
- [x] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ).
- [x] If your PR is related to the `recipe` submodule, please also update the reference to the submodule commit via `git submodule update --remote` or `cd recipe && git pull origin main`.

### CODE REVIEW

**Comment by wuxibin89** [comment]
legacy worker(fsdp_workers.py/megatron_workers.py) will soon be deprecated in v0.7.2, please use `engine_workers.py` by
```
trainer.use_legacy_worker_impl=disable
```

**Comment by HollowMan6** [comment]
@cavities12 I see you have already include the changes here in https://github.com/verl-project/verl/pull/5535 right? Do you still want to keep this PR? For me I think it would be better to merge this PR to https://github.com/verl-project/verl/pull/5535 if dependent.

**Comment by cavities12** [comment]
> @cavities12 I see you have already include the changes here in #5535 right? Do you still want to keep this PR? For me I think it would be better to merge this PR to #5535 if dependent.

Good call, will close this one
